sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -82,6 +82,14 @@ abstract class JdbcDialect {
   def getJDBCType(dt: DataType): Option[JdbcType] = None
 
   /**
+   * Retrieve the jdbc / sql type for a given datatype.
+   * @param dt The datatype (e.g. [[org.apache.spark.sql.types.StringType]])
+   * @param md The metadata
+   * @return The new JdbcType if there is an override for this DataType
+   */
+  def getJDBCType(dt: DataType, md: Metadata): Option[JdbcType] = None
```

Comment:
How about calling `getJDBCType(dt,Metadata.empty)` instead of `None` in `getJDBCType(dt: DataType)`, because the classes that extends `JdbcDialect` possibly implement one of them and then the behaviours of the two function totally different?